github/projects.go

```diff
@@ -216,3 +213,293 @@ func (s *ProjectsService) ListProjectFieldsForOrg(ctx context.Context, org strin
 	}
 	return fields, resp, nil
 }
+
+// ListProjectFieldsForUser lists Projects V2 for a user.
+//
+// GitHub API docs: https://docs.github.com/rest/projects/fields#list-project-fields-for-user
+//
+//meta:operation GET /users/{username}/projectsV2/{project_number}/fields
+func (s *ProjectsService) ListProjectFieldsForUser(ctx context.Context, user string, projectNumber int, opts *ListProjectsOptions) ([]*ProjectV2Field, *Response, error) {
+	u := fmt.Sprintf("users/%v/projectsV2/%v/fields", user, projectNumber)
+	u, err := addOptions(u, opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var fields []*ProjectV2Field
+	resp, err := s.client.Do(ctx, req, &fields)
+	if err != nil {
+		return nil, resp, err
+	}
+	return fields, resp, nil
+}
+
+// GetProjectFieldForOrg gets a single project field from an organization owned project.
+//
+// GitHub API docs: https://docs.github.com/rest/projects/fields#get-project-field-for-organization
+//
+//meta:operation GET /orgs/{org}/projectsV2/{project_number}/fields/{field_id}
+func (s *ProjectsService) GetProjectFieldForOrg(ctx context.Context, org string, projectNumber int, fieldID int64) (*ProjectV2Field, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/projectsV2/%v/fields/%v", org, projectNumber, fieldID)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	field := new(ProjectV2Field)
+	resp, err := s.client.Do(ctx, req, field)
+	if err != nil {
+		return nil, resp, err
+	}
+	return field, resp, nil
+}
+
+// GetProjectFieldForUser gets a single project field from a user owned project.
+//
+// GitHub API docs: https://docs.github.com/rest/projects/fields#get-project-field-for-user
+//
+//meta:operation GET /users/{username}/projectsV2/{project_number}/fields/{field_id}
+func (s *ProjectsService) GetProjectFieldForUser(ctx context.Context, user string, projectNumber int, fieldID int64) (*ProjectV2Field, *Response, error) {
```

Comment:
```suggestion
func (s *ProjectsService) GetUserProjectField(ctx context.Context, user string, projectNumber int, fieldID int64) (*ProjectV2Field, *Response, error) {
```

See #3761

Reply:
refactored as part of c2f8bfadc2701c3504de4e4d05e623d478899dba